source/common/crypto/utility_impl.cc

```diff
@@ -35,6 +35,15 @@ std::vector<uint8_t> UtilityImpl::getSha256Hmac(const std::vector<uint8_t>& key,
   return hmac;
 }
 
+std::vector<uint8_t> UtilityImpl::getSha256Hmac(const std::vector<uint8_t>& key,
```

Comment:
is it possible to reuse above function?

Reply:
i don't think we can. The current uses of getSha256Hmac are on string based payloads. The existing getSha256Hmac does a conversion from absl::string_view to uint8_t*. Whereas the one I've added is based on a binary payload that does not require conversion.

To reuse would require converting to a string_view, which is not correct for a binary payload.

If I was going to tackle this correctly, I'd likely remove the string version entirely and get the consumers to convert to uint8 array before calling. This is because sha256hmac is implicitly a binary function. If the only consumer was the request signing filter, I'd do this, but unfortunately the string version is also in use by the oauth2 filter, which I'd rather not touch.

Reply:
In fact, the absl::string_view or std::string in C++ basically could be treated as byte array.

And if you insist to add this new method, I think absl::Span<uint8_t> would be better than `std::vector<uint8_t>`, because the user of this method may don't want to create a `std::vector<uint_8>`.

Reply:
have removed this additional function and implemented via reinterpret_cast instead. if you're ok with that then so am I